src/components/fabSpeedDial/fabSpeedDial.spec.js

```diff
@@ -0,0 +1,95 @@
+describe('<md-fab-speed-dial> directive', function() {
+
+  beforeEach(module('material.components.fabSpeedDial'));
+
+  function setup(template) {
+    var el;
+
+    inject(function($compile, $rootScope) {
+      el = $compile(template)($rootScope.$new());
+      $rootScope.$apply();
+    });
+
+    return el;
+  }
+
+  it('applies a class for each direction', inject(function() {
+    var fab = setup([
+      '<md-fab-speed-dial md-direction="{{direction}}"></md-fab-speed-dial>'
+    ].join("\n"));
+
+    var scope = fab.scope();
+
+    scope.$apply('direction = "down"');
+    expect(fab.hasClass('md-down')).toBe(true);
+
+    scope.$apply('direction = "up"');
+    expect(fab.hasClass('md-up')).toBe(true);
+
+    scope.$apply('direction = "left"');
+    expect(fab.hasClass('md-left')).toBe(true);
+
+    scope.$apply('direction = "right"');
+    expect(fab.hasClass('md-right')).toBe(true);
+  }));
+
+  it('opens when the trigger element is focused', inject(function() {
+    var fab = setup([
+      '<md-fab-speed-dial><md-fab-trigger><button></button></md-fab-trigger></md-fab-speed-dial>'
+    ].join('\n'));
+
+    fab.find('button').triggerHandler('focus');
+    expect(fab.hasClass('md-opened')).toBe(true);
+  }));
+
+  it('opens when the speed dial elements are focused', inject(function() {
+    var fab = setup([
+      '<md-fab-speed-dial><md-fab-actions><button></button></md-fab-actions></md-fab-speed-dial>'
+    ].join('\n'));
+
+    fab.find('button').triggerHandler('focus');
+    expect(fab.hasClass('md-opened')).toBe(true);
+  }));
+
+  it('closes when the speed dial elements are blurred', inject(function($timeout) {
+    var fab = setup([
+      '<md-fab-speed-dial><md-fab-actions><button></button></md-fab-actions></md-fab-speed-dial>'
+    ].join('\n'));
+
+    fab.find('button').triggerHandler('focus');
+    $timeout(function() {
```

Comment:
The inside of a `$timeout` doesn't run in unit tests- `$timeout` is replaced with a mock service. 

Instead, call `$timeout.flush()` and then make your assertion.